Classes/Middleware/RequestCacheMiddleware.php

```diff
@@ -127,13 +139,17 @@ protected function getCacheIdentifierForRequestIfCacheable(ServerRequestInterfac
         $requestQueryParams = $request->getQueryParams();
         $allowedQueryParams = [];
         $ignoredQueryParams = [];
+        $respectedQueryParams = [];
         $disallowedQueryParams = [];
         foreach ($requestQueryParams as $key => $value) {
             switch (true) {
                 case (in_array($key, $this->allowedQueryParams)):
                     $allowedQueryParams[$key] = $value;
                     break;
-                case (in_array($key, $this->ignoredQueryParams)):
+                case (in_array($key, $this->ignoredQueryParams) && empty($this->respectedQueryParams)):
+                    $ignoredQueryParams[$key] = $value;
+                    break;
+                case (!in_array($key, $this->respectedQueryParams) && empty($this->ignoredQueryParams)):
```

Comment:
I think a respected queryParameter should be added to `$allowedQueryParams[$key]` as i would assume this to be part of the cache identity. 

That imho would also mean that allowedQueryParams should only be checked when $this->respectedQueryParams is empty.

Reply:
@mficzel How would you configure FullPageCache to not cache a search query in this case?